### DISCUSSION TITLE
Error of "Invoke-PnPSiteTemplate”

### DISCUSSION THREAD

**Discussion by IobeHisashi** [pnp/powershell#2527] [Q&A]
As you can see below, I have created a site template in Script and want to reflect the template on another site.
When I run the Script, I get an error.
Please let me know the cause and how to deal with it.

■Get-PnPSiteTemplate
$TemplatePath = "C:\SharePointOnlinePowerShell\CopySite\template.pnp"
Get-PnPSiteTemplate -Out $TemplatePath -IncludeAllPages -PersistBrandingFiles -PersistPublishingFiles 
Add-PnPFileToSiteTemplate -Path $TemplatePath -SourceUrl "SitePages/Home.aspx"

■Invoke-PnPSiteTemplate
$TemplatePath = "C:\SharePointOnlinePowerShell\CopySite\template.pnp"
Invoke-PnPSiteTemplate -Path $TemplatePath -ClearNavigation -ProvisionContentTypesToSubWebs -ProvisionFieldsToSubWebs

■Error Message（Original:Japanese)
Invoke-PnPSiteTemplate : テンプレートの URL が正しくありません。
テンプレートは、このドキュメント ライブラリの Forms ディレクトリに保存されている必要があります。Forms ディレクトリにテンプレートを作成して、Web アドレスを再入力してください。Forms ディレクトリにテンプレートを移動またはコピーすることはできません。
発生場所 C:\SharePointOnlinePowerShell\CopySite\CopySite3.ps1:40 文字:1
+ Invoke-PnPSiteTemplate -Path $TemplatePath -ClearNavigation -Provisio ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Invoke-PnPSiteTemplate], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.InvokeSiteTemplate

■Error Message（Translation：English)
Invoke-PnPSiteTemplate : The URL for the template is incorrect.
The template must be stored in the Forms directory of this document library. create the template in the Forms directory and re-enter the web address. the template cannot be moved or copied to the Forms directory.

**Comment by AndersRask** [comment]
Try and enable trace and check it for clues

`Set-PnPTraceLog -On -Level Information -LogFile tracelog.txt`

Remember to set it to -Off agan or close the shell

**Comment by IobeHisashi** [comment]
Dear Sir,

I send trace log.
Please check.

Regards,
MITSUBA
Iobe Hisashi

**Comment by AndersRask** [comment]
Try and split up the provisioning template in seperate handlers and run them seperately to get a better view of what is going on.
Eg. columns, content types, lists

**Comment by IobeHisashi** [comment]
Dear Rask san,

What is the specific method of confirmation?

Regards,
MITSUBA
Iobe Hisashi

**Comment by AndersRask** [comment]
Just split up template.xml into different handlers to isolate where the issue is:

Invoke-PnPSiteTemplate -Path columns.xml -Handlers Fields
then
Invoke-PnPSiteTemplate -Path contentTypes.xml -Handlers ContentTypes
then
Invoke-PnPSiteTemplate -Path lists.xml -Handlers Lists

You can also try and strip the list template for all unnecessary clutter that was added during export.
Things like FieldRefs, Webhooks, FieldDefaults to isolate the issue

**Comment by IobeHisashi** [comment]
Dear Rask san,

I was able to confirm that the error occurs in List.
I send a trace log.
What should I do for errors after this?

Regards,
MITSUBA
Iobe Hisashi

**Reply by AndersRask** [reply]
Sorry, my japaneese is a bit rusty :)

What I usually do if the trace log doesnt help, is to try and remove pieces of the list template until it works, then work yourself backwards and fix the issues.

**Comment by IobeHisashi** [comment]
Dear Rask san,

I have checked the library template settings.
The template file exists.
I do not use content types, so there is only one template file.
Please check the attached file for details.

Regards,
MITSUBA
Iobe Hisashi

**Comment by IobeHisashi** [comment]
Dear Rask san,

I changed the file template settings as follows and the error was resolved.

ExData/Forms/template.doc -> ExData/Forms/template.dotx

Are you sure that the Invoke-PnPSiteTemplate command will cause an error if template.doc is set as the file template?

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 10:39 AM
To: 'pnp/powershell' ***@***.***>; pnp/powershell ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I have checked the library template settings.
The template file exists.
I do not use content types, so there is only one template file.
Please check the attached file for details.

Regards,
MITSUBA
Iobe Hisashi

**Comment by IobeHisashi** [comment]
Dear Rask san,

I was able to resolve this error, so I created a new site and ran the process.
A new error occurred.

Error:
Invoke-PnPSiteTemplate : Unknown error
k生鏊 C:\SharePointOnlinePowerShell\Copy\CopySite\test1\CopySite3.ps1:43 文字:1
+ Invoke-PnPSiteTemplate -Path $TemplatePath -ClearNavigation -Provisio ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Invoke-PnPSiteTemplate], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.InvokeSiteTemplate

I have confirmed that an error is occurring with the command Invoke-PnPSiteTemplate -Handlers Lists.
What should I do to resolve the error?

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 11:16 AM
To: 'pnp/powershell' ***@***.***>; 'pnp/powershell' ***@***.***>
Cc: 'Author' ***@***.***>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I changed the file template settings as follows and the error was resolved.

ExData/Forms/template.doc -> ExData/Forms/template.dotx

Are you sure that the Invoke-PnPSiteTemplate command will cause an error if template.doc is set as the file template?

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 10:39 AM
To: 'pnp/powershell' ***@***.******@***.***>>; pnp/powershell ***@***.******@***.***>>
Cc: Author ***@***.******@***.***>>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I have checked the library template settings.
The template file exists.
I do not use content types, so there is only one template file.
Please check the attached file for details.

Regards,
MITSUBA
Iobe Hisashi

**Comment by IobeHisashi** [comment]
Dear Rask san,

I have self-resolved this error.

Error:
Invoke-PnPSiteTemplate : Unknown error

Please reply only about the following

ExData/Forms/template.doc -> ExData/Forms/template.dotx

Are you sure that the Invoke-PnPSiteTemplate command will cause an error if template.doc is set as the file template?




From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 11:49 AM
To: 'pnp/powershell' ***@***.***>; 'pnp/powershell' ***@***.***>
Cc: 'Author' ***@***.***>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I was able to resolve this error, so I created a new site and ran the process.
A new error occurred.

Error:
Invoke-PnPSiteTemplate : Unknown error
k生鏊 C:\SharePointOnlinePowerShell\Copy\CopySite\test1\CopySite3.ps1:43 文字:1
+ Invoke-PnPSiteTemplate -Path $TemplatePath -ClearNavigation -Provisio ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Invoke-PnPSiteTemplate], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.InvokeSiteTemplate

I have confirmed that an error is occurring with the command Invoke-PnPSiteTemplate -Handlers Lists.
What should I do to resolve the error?

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 11:16 AM
To: 'pnp/powershell' ***@***.******@***.***>>; 'pnp/powershell' ***@***.******@***.***>>
Cc: 'Author' ***@***.******@***.***>>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I changed the file template settings as follows and the error was resolved.

ExData/Forms/template.doc -> ExData/Forms/template.dotx

Are you sure that the Invoke-PnPSiteTemplate command will cause an error if template.doc is set as the file template?

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 10:39 AM
To: 'pnp/powershell' ***@***.******@***.***>>; pnp/powershell ***@***.******@***.***>>
Cc: Author ***@***.******@***.***>>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I have checked the library template settings.
The template file exists.
I do not use content types, so there is only one template file.
Please check the attached file for details.

Regards,
MITSUBA
Iobe Hisashi

**Comment by IobeHisashi** [comment]
Dear Rask san,

I have an additional question.
The following error occurred.

■エラ`
Invoke-PnPSiteTemplate : このフィ`ルドを更新できません。
Invoke-PnPSiteTemplate : Unable to update this field.　※Machine Translation

I checked the trace log, but I cannot determine which field it is.
Is there any way to find out?
I will send you the trace log.

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Wednesday, November 16, 2022 8:54 AM
To: 'pnp/powershell' ***@***.***>; 'pnp/powershell' ***@***.***>
Cc: 'Author' ***@***.***>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I have self-resolved this error.

Error:
Invoke-PnPSiteTemplate : Unknown error

Please reply only about the following

ExData/Forms/template.doc -> ExData/Forms/template.dotx

Are you sure that the Invoke-PnPSiteTemplate command will cause an error if template.doc is set as the file template?




From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 11:49 AM
To: 'pnp/powershell' ***@***.******@***.***>>; 'pnp/powershell' ***@***.******@***.***>>
Cc: 'Author' ***@***.******@***.***>>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I was able to resolve this error, so I created a new site and ran the process.
A new error occurred.

Error:
Invoke-PnPSiteTemplate : Unknown error
k生鏊 C:\SharePointOnlinePowerShell\Copy\CopySite\test1\CopySite3.ps1:43 文字:1
+ Invoke-PnPSiteTemplate -Path $TemplatePath -ClearNavigation -Provisio ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Invoke-PnPSiteTemplate], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.InvokeSiteTemplate

I have confirmed that an error is occurring with the command Invoke-PnPSiteTemplate -Handlers Lists.
What should I do to resolve the error?

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 11:16 AM
To: 'pnp/powershell' ***@***.******@***.***>>; 'pnp/powershell' ***@***.******@***.***>>
Cc: 'Author' ***@***.******@***.***>>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I changed the file template settings as follows and the error was resolved.

ExData/Forms/template.doc -> ExData/Forms/template.dotx

Are you sure that the Invoke-PnPSiteTemplate command will cause an error if template.doc is set as the file template?

Regards,
MITSUBA
Iobe Hisashi


From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Tuesday, November 15, 2022 10:39 AM
To: 'pnp/powershell' ***@***.******@***.***>>; pnp/powershell ***@***.******@***.***>>
Cc: Author ***@***.******@***.***>>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I have checked the library template settings.
The template file exists.
I do not use content types, so there is only one template file.
Please check the attached file for details.

Regards,
MITSUBA
Iobe Hisashi

**Reply by AndersRask** [reply]
I have no idea about the template, i would suggest trying different options and see what works.

About updating a field, it can be troublesome to update existing fields if they are already used. The best option is often to hide a field and add a new one, if you want to make changes to fields within a list or a content type.

**Comment by IobeHisashi** [comment]
Dear Rask san,

Could you please elaborate a bit more on this procedure?

The best option is often to hide a field and add a new one, if you want to make changes to fields within a list or a content type.

**Comment by AndersRask** [comment]
Your error was machine translated, so just guessed that you were updating existing fields.
That might be ok, but if the field you are adding are of another type as the existing, or different properties, this could give you errors.

As with all debugging, try to isolate the issue:
-find the column that gives you issues
-determine what the issue is by comparing existing field in the site with the template
-fix using a different internal name, or removing the existing field first
-the hiding part is mainly if you use content types, which complicates things, but if you are not using those then ignore that advice

Venlig hilsen
Anders Rask

Fra: IobeHisashi ***@***.***>
Sendt: Wednesday, 16 November 2022 09.45
Til: pnp/powershell ***@***.***>
Cc: Anders Rask ***@***.***>; Comment ***@***.***>
Emne: Re: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

Could you please elaborate a bit more on this procedure?

The best option is often to hide a field and add a new one, if you want to make changes to fields within a list or a content type.

**Comment by IobeHisashi** [comment]
Dear Rask san,

I have confirmed that the following error occurs in -Handlers Files.
What could you mean by a field error in Files?

■Error
Invoke-PnPSiteTemplate : このフィ`ルドを更新できません。
Invoke-PnPSiteTemplate : Unable to update this field.

Regards,
MITSUBA
Iobe Hisashi

From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Wednesday, November 16, 2022 5:44 PM
To: 'pnp/powershell' ***@***.***>; pnp/powershell ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

Could you please elaborate a bit more on this procedure?

The best option is often to hide a field and add a new one, if you want to make changes to fields within a list or a content type.

**Comment by AndersRask** [comment]
Try and post the field.xml

Doesn’t the PnP Trace file give a hint on what field that is giving trouble?

Else try this:

$error[0] | fl * -force

To see error details

Venlig hilsen
Anders Rask

Fra: IobeHisashi ***@***.***>
Sendt: Thursday, 17 November 2022 07.15
Til: pnp/powershell ***@***.***>
Cc: Anders Rask ***@***.***>; Comment ***@***.***>
Emne: Re: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

I have confirmed that the following error occurs in -Handlers Files.
What could you mean by a field error in Files?

■Error
Invoke-PnPSiteTemplate : このフィ`ルドを更新できません。
Invoke-PnPSiteTemplate : Unable to update this field.

Regards,
MITSUBA
Iobe Hisashi

From: MCJ-IOBE-HISASHI 五百部 寿史
Sent: Wednesday, November 16, 2022 5:44 PM
To: 'pnp/powershell' ***@***.***>; pnp/powershell ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [pnp/powershell] Error of "Invoke-PnPSiteTemplate” (Discussion #2527)

Dear Rask san,

Could you please elaborate a bit more on this procedure?

The best option is often to hide a field and add a new one, if you want to make changes to fields within a list or a content type.

**Comment by IobeHisashi** [comment]
Dear Rask san,

Thanks for the reply.

I have another error.
The error is that seattle.master cannot be moved.
How can i work around this?

Log
＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝
powershell_ise.exe Information: 0 : 2022-11-18 16:41:05.3251 [Files]   [16]     [Debug]           Uploading and overwriting existing file seattle.master      1231ms 5734b729-7f03-45d7-9355-7bc96723b6b8
powershell_ise.exe Information: 0 : 2022-11-18 16:41:05.4501 [PnP.Framework]          [0]           [Information]    File seattle.master retrieved from folder _catalogs\masterpage   0ms
powershell_ise.exe Error: 0 : 2022-11-18 16:41:05.7313          [PnP.Framework]          [0]           [Error]  ExecuteQuery threw following exception: Microsoft.SharePoint.Client.ServerException: このファイルを移印⑾鞒⒚涓蓼郡暇集することはできません。
   鏊 Microsoft.SharePoint.Client.ClientRequest.ProcessResponseStream(Stream responseStream)
   鏊 Microsoft.SharePoint.Client.ClientRequest.ProcessResponse()
   鏊 Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryToServerAsync>d__53.MoveNext()
--- 直前に例外がスロ`された鏊椁违攻骏氓 トレ`スのKわり ---
   鏊 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   鏊 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   鏊 Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryAsync>d__39.MoveNext()
--- 直前に例外がスロ`された鏊椁违攻骏氓 トレ`スのKわり ---
   鏊 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   鏊 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   鏊 Microsoft.SharePoint.Client.ClientRuntimeContext.<ExecuteQueryAsync>d__65.MoveNext()
--- 直前に例外がスロ`された鏊椁违攻骏氓 トレ`スのKわり ---
   鏊 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   鏊 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   鏊 Microsoft.SharePoint.Client.ClientContext.<ExecuteQueryAsync>d__28.MoveNext()
--- 直前に例外がスロ`された鏊椁违攻骏氓 トレ`スのKわり ---
   鏊 System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   鏊 System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   鏊 Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__6.MoveNext()
ServerErrorCode: -2130575223
ServerErrorTypeName: Microsoft.SharePoint.SPException
ServerErrorTraceCorrelationId: bfbd79a0-304a-1000-d623-1779e125970a
ServerErrorValue:
ServerErrorDetails:
.          0ms
powershell_ise.exe Information: 0 : 2022-11-18 16:41:06.3563 [Files]   [16]     [Debug]           Uploading and overwriting existing file v4.master 2263ms 5734b729-7f03-45d7-9355-7bc96723b6b8
＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝＝
